module/system_power/src/mod_system_power.c

```diff
@@ -408,7 +408,6 @@ static int system_power_mod_init(fwk_id_t module_id,
     const struct mod_system_power_config *config;
 
     fwk_assert(data != NULL);
-    fwk_check(element_count > 0);
```

Comment:
`element_count == 0` would leave to unexpected behavior on `fwk_mm_calloc` call on line 418.  And since the module is not expecting to have zero elements some parts of the code need to be reworked